src/string/utf16.cr

```diff
@@ -81,6 +73,41 @@ class String
     end
   end
 
+  # Decodes the given *slice* UTF-16 sequence into a String and returns the
+  # pointer after reading. The string ends when a zero value is found.
+  #
+  # ```
+  # slice = Slice[104_u16, 105_u16, 0_u16, 55296_u16, 56485_u16, 0_u16]
+  # String.from_utf16(slice) # => "hi\0000𐂥"
+  # pointer = slice.to_unsafe
+  # string, pointer = String.from_utf16(pointer) # => "hi"
+  # string, pointer = String.from_utf16(pointer) # => "𐂥"
+  # ```
+  #
+  # Invalid values are encoded using the unicode replacement char with
+  # codepoint `0xfffd`.
+  def self.from_utf16(pointer : Pointer(UInt16)) : {String, Pointer(UInt16)}
```

Comment:
This is a breaking change and as such needs some more discussion.

Reply:
Sure. When reading from a pointer it needs to advance the amount of `UInt16` read. We can't determine that afterwards from `String#bytesize` because UTF-8 and UTF-16 have different bytesizes. We could rename the method with this changed behaviour (to maybe `String.from_utf16_advancing_pointer`). Or change the signature in some way. But I don't think it makes much sense to have both a version of this method (with pointer) that returns the advanced pointer and one that doesn't. It's easy to just ignore the returned pointer value if you don't need it.